openapi3/components_test.go

```diff
@@ -0,0 +1,37 @@
+package openapi3
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestValidateIdentifier(t *testing.T) {
+	tt := []struct {
+		name       string
+		identifier string
+		isValid    bool
+		errMsg     string
+	}{
+		{name: "valid", identifier: "User-12_3.4", isValid: true, errMsg: ""},
+		{name: "valid with []", identifier: "User[Admin]", isValid: true, errMsg: ""},
+		{name: "invalid", identifier: "User Admin", isValid: false, errMsg: ""},
```

Comment:
Looks like errMsg is never not empty.
Instead of all this, how about just:
```go
	for identifier, isValid := range map[string]bool{
		"User-12_3.4": true,
		"User[Admin]": true,
		"User Admin": false,
	}{

```